internal/main.go

```diff
@@ -0,0 +1,238 @@
+	Genres        []struct {
+		Name string `json:"name" gorm:"type:text"`
+	} `json:"genres" gorm:"-"`
+	PosterPath string   `json:"poster_path"`
+	Popularity float32  `json:"popularity"`
+	Providers  Provider `json:"watch/providers"`
+}
+
+func (tv *TV) toMedia() *DBMedia {
+	genres := []string{}
+	for _, genre := range tv.Genres {
+		genres = append(genres, genre.Name)
+	}
+	return &DBMedia{
+		Id:            "t" + strconv.Itoa(tv.Id),
+		Title:         tv.Name,
+		OriginalTitle: tv.OriginalTitle,
+		Overview:      tv.Overview,
+		ReleaseDate:   tv.FirstAirDate,
+		Genres:        genres,
+		PosterPath:    tv.PosterPath,
+		Popularity:    tv.Popularity,
+		Providers:     tv.Providers,
+	}
+}
+
+type Provider struct {
+	Results map[string]struct {
+		Flatrate []struct {
+			DisplayPriority int    `json:"display_priority"`
+			LogoPath        string `json:"logo_path"`
+			Id              int    `json:"provider_id"`
+			Name            string `json:"provider_name"`
+		} `json:"flatrate,omitempty"`
+		Free []struct {
+			DisplayPriority int    `json:"display_priority"`
+			LogoPath        string `json:"logo_path"`
+			Id              int    `json:"provider_id"`
+			Name            string `json:"provider_name"`
+		} `json:"free,omitempty"`
+	} `json:"results" gorm:"type:json"`
+}
+
+type MediaIds struct {
+	Ids []string `json:"ids"`
+}
+
+type Env struct {
+	db *gorm.DB
+}
+
+var TMDB_KEY = os.Getenv("TMDB_KEY")
+
+func main() {
+	dsn := "host=db user=postgres password=postgres dbname=streamchaser port=5432 sslmode=disable"
+	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	if err != nil {
+		panic("Could not connect to DB!")
+	}
+	db.AutoMigrate(&DBMedia{})
+
+	if TMDB_KEY == "" {
+		panic("No TMDB key provided")
+	}
+
+	env := &Env{db: db}
+
+	router := gin.Default()
+	router.POST("/update-media", env.processIds)
+	router.Run(":8888")
+}
+
+func (e *Env) processIds(c *gin.Context) {
+	media := MediaIds{}
+
+	c.Bind(&media)
+	if len(media.Ids) == 0 {
+		c.JSON(http.StatusBadRequest, gin.H{"info": "Please provide id's to fetch"})
+		return
+	}
+
+	var wg sync.WaitGroup
+	movieCh := make(chan Movie, len(media.Ids))
+	tvCh := make(chan TV, len(media.Ids))
+	guardCh := make(chan int, 100)
+	dbMedia := []DBMedia{}
+	for _, id := range media.Ids {
+		guardCh <- 1
+		wg.Add(1)
+		switch string([]rune(id)[0]) {
+		case "m":
+			go func(id string, movieCh chan Movie) {
+				defer wg.Done()
+				fetchMovie(id[1:], movieCh)
+				<-guardCh
+			}(id, movieCh)
+		case "t":
+			go func(id string, TVCh chan TV) {
+				defer wg.Done()
+				fetchTV(id[1:], TVCh)
+				<-guardCh
+			}(id, tvCh)
+		default:
+			log.Fatal("This is not the movie type you seek")
+		}
+	}
+	wg.Wait()
+	close(movieCh)
+	close(tvCh)
+
+	for tv := range tvCh {
+		if tv.Popularity > 1 {
+			dbMedia = append(dbMedia, *tv.toMedia())
+		}
+	}
+	for movie := range movieCh {
+		if movie.Popularity > 1 {
+			dbMedia = append(dbMedia, *movie.toMedia())
+		}
+	}
```

Comment:
Can we take `Popularity` as an optional query param and default to 0?

Reply:
Fuck it, just add a TODO :D

Reply:
Done